images/build/debian-iptables/distroless/Dockerfile

```diff
@@ -0,0 +1,54 @@
+# Copyright 2022 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+ARG GORUNNERIMAGE
+# Using directly debian base, as we are going to just copy the required binaries
+FROM debian:bullseye as build
+
+COPY stage-binary-and-deps.sh /
+COPY get-binaries-from-package.sh /
+
+ARG STAGE_DIR="/opt/stage"
+RUN apt -y update && \
+    apt -y install bash && \
+    mkdir -p "${STAGE_DIR}" && \
+    /get-binaries-from-package.sh conntrack "${STAGE_DIR}" && \
+    /get-binaries-from-package.sh ebtables "${STAGE_DIR}" && \
+    /get-binaries-from-package.sh ipset "${STAGE_DIR}" && \
+    /get-binaries-from-package.sh iptables "${STAGE_DIR}" && \
+    /get-binaries-from-package.sh kmod "${STAGE_DIR}" && \
+    /stage-binary-and-deps.sh /bin/dash "${STAGE_DIR}" && \
+    /stage-binary-and-deps.sh /bin/grep "${STAGE_DIR}" && \
+    /stage-binary-and-deps.sh /usr/bin/wc "${STAGE_DIR}" && \
```

Comment:
wc is used by iptables-wrapper

Reply:
I think it might be worth a comment in the Dockerfile explaining why for each of these? Either a block comment above or the:
```
`# comment` \
```
hack.

Reply:
huh... I didn't even think about trying to limit the use of external tools...

We could trivially replace `wc -l` with `grep -c .`
I don't think we can easily get rid of `grep`.

I don't know how much space the `update-alternatives` framework takes up, but the upstream version of the wrapper script will work even without it. (It will just redo the symlinks itself)

Reply:
Ah ok, I will check with both update-alternatives there and not. I don't think wc is that big we should replace for all grep -c, but up to you.

Let me work on using the installer directly and watching how it goes.